save/delete documents in batch

matsunanaro opened the pull request:
This PR implemented two new methods to save and delete documents in batch. These methods are making use of the inBatch method of its underlying Swift and Java Couchbase Lite SDK. Calling these methods are more efficient than saving/deleting document on an individual basis. 

The following methods are added to the Database class:

1. saveDocuments
  /// Saves documents in batch.
  /// 
  /// Returns no. of successfully inserted documents 
  /// 
  /// Note: It's more efficient to call this method to save
  /// documents in batch than calling [saveDocument] on each document.
  Future<int> saveDocuments(List<MutableDocument> docs,
      {ConcurrencyControl concurrencyControl =
          ConcurrencyControl.lastWriteWins})

2. deleteDocuments

    /// Deletes documents in batch 
    /// 
    /// [ids] - the ids of the documents to be deleted.
    /// 
    /// Returns no. of successfully deleted documents.
    /// 
    /// Note: it's more efficient to call this method to delete documents than
    /// than calling deletDocument on each document.
    Future<int> deleteDocuments(List<String> ids)

matsunanaro committed:
save/delete docs in batch

matsunanaro committed:
Added tests for deleteDocuments and saveDocuments

Commit:
saves/seletes documents asynchronously

matsunanaro commented:
The check failure seems to be coming from CI, which has something to do with the inclusion of the "Podfile.lock" file.
Any idea how to fix that?

bawelter commented:
The error is from the publish dry run.  Which means there was probably no notes added for the release or the version was not bumped or both.  I believe you should be able to do the dry run locally to ensure it's all good before pushing the changes.

Commit:
changed version to 3.0.0-nullsafety.3

Commit:
fixed Pub  latest error

Commit:
Revert "fixed Pub  latest error"

This reverts commit 14ebcbe1212ca46d631e1a87084ed8bea6cb92df.

Commit:
do not publish Podfile.lock

The inclusion of Podfile.lock was complained by the latest Pub

Commit:
Revert "do not publish Podfile.lock"

This reverts commit d04fc4741901909d0bd6439554c29db7c13bf713.

Commit:
added back Podfile.lock to gitignore

Commit:
Deleted Podfile.lock from repo

matsunanaro commented:
> The error is from the publish dry run. Which means there was probably no notes added for the release or the version was not bumped or both. I believe you should be able to do the dry run locally to ensure it's all good before pushing the changes.

Thanks for the tip. I tried your suggestion and added the release no. But no error was shown when I ran "dry run" locally.
But the error appeared again when I pushed my commit.

It seemed that the latest Pub module does not allow publishing Podfile.lock (due to some reason)

I had to remove the Podfile.lock  from the repo by "git rm --cached example/ios/Podfile.lock".  
In addition, I added it back to .gitignore to prevent the file from committing to the repo in future commits.

Thanks again.
